src/api-binder/report.ts

```diff
@@ -141,21 +145,45 @@ export async function startReporting() {
 	])) as StateReportOpts;
 
 	let reportPending = false;
-	const doReport = async () => {
+	// Reports current state if not already sending and does not exceed report
+	// frequency. Returns true if sent; otherwise false.
+	const doReport = async (): boolean => {
 		if (!reportPending) {
-			reportPending = true;
-			await reportCurrentState(reportConfigs);
-			reportPending = false;
+			const sinceLastReport = performance.now() - lastReportTime;
+			if (sinceLastReport > constants.maxReportFrequency) {
+				reportPending = true;
+				await reportCurrentState(reportConfigs);
+				reportPending = false;
+				return true;
+			} else {
+				return false;
+			}
+		} else {
+			return false;
 		}
 	};
 
+	const onStateChange = async () => {
+		// Defers to timed report schedule if we can't report immediately.
+		// Ensures that we always report on a change event.
+		stateChangeDeferred = ! await doReport();
+	}
+
 	// If the state changes, report it
-	deviceState.on('change', doReport);
+	deviceState.on('change', onStateChange);
 
 	async function recursivelyReport(delayBy: number) {
 		try {
-			// Try to send current state
-			await doReport();
+			// Follow-up when report not sent immediately on change event...
+			if (stateChangeDeferred) {
+				stateChangeDeferred = ! await doReport();
```

Comment:
```suggestion
				stateChangeDeferred = !(await doReport());
```
I would format both instances of this as the above to be inline with the rest of the codebase.